### DISCUSSION TITLE
Flashing command

### DISCUSSION THREAD

**Discussion by markpeletier** [sle118/squeezelite-esp32#129] [Q&A] [answered]
At #building-squeezelite-esp32 the following flashing command is given:

    <path_to_your_python>/python.exe <path_to_your_esptool>/esptool.py -p %ESPPORT% -b %ESPBAUD% --before default_reset --after hard_reset write_flash --flash_mode dio --flash_size detect --flash_freq 80m 0x150000 build/squeezelite.bin

If I try this on a WROVER that had previously been flashed with the Arduino IDE with a standard Blink example, then the ESP boots into the Blink example. 

The Blink example has an esptool command line that contains four parts: two boot loader files, a partition list, and the Blink executable itself (at least, this is what I think they are). Also the example at https://docs.espressif.com/projects/esptool/en/latest/esp32s3/esptool/flashing-firmware.html#flashing shows multiple parts (three, in that case). 

How should this work in the case of squeezelite? Is the command line above missing some parts?

**Comment by Joshfindit** [comment, marked as answer]
It's not visible in the README, but there is more detail here: https://forums.slimdevices.com/showthread.php?112697-ANNOUNCE-Squeezelite-ESP32-(dedicated-thread)

There are two notable things:
1. The ESP used must have 4MB or RAM *and* at least 4MB of SRAM. The post mentions that WROOM modules are no good, and WROVER must be used instead. `esptool` does not seem to throw an error if it fills the module's RAM and there's still more file to write.
2. The command is now `python ${IDF_PATH}/components/esptool_py/esptool/esptool.py --chip esp32 --port ${ESPPORT} --baud ${ESPBAUD} --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size detect 0xd000 ota_data_initial.bin 0x1000 bootloader.bin 0x10000 recovery.bin 0x8000 partitions.bin` (Done from withing the `build` folder inside the zipped release)

And an update: As of #144, there is a web installer that does not need any of the `esptool` commands

**Reply by sle118** [reply]
You are correct. The web installer can be used to flash the initial firmware and from there, updates will be performed via the web interface

**Comment by dothebart** [comment]
It seems the directory layout of the zip changed meanwhile, It seems this is working for me:
```
 /usr/local/bin/esptool.py --chip esp32 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size detect 0xd000 ota_data_initial.bin 0x1000 bootloader/bootloader.bin 0x10000 recovery.bin 0x8000 partition_table/partition-table.bin 
```
